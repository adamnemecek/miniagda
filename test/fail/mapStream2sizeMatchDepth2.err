MiniAgda by Andreas Abel and Karl Mehltretter
--- opening "mapStream2sizeMatchDepth2.ma" ---
--- scope checking ---
--- type checking ---
type  Stream : ++(A : Set) -> - Size -> Set
term  cons : .[A : Set] -> .[i : Size] -> ^(y1 : A) -> ^(y2 : Stream A i) -> < cons i y1 y2 : Stream A $i >
type  Nat : Set
term  zero : < zero : Nat >
term  succ : ^(y0 : Nat) -> < succ y0 : Nat >
term  map2 : .[i : Size] -> (Nat -> Nat) -> Stream Nat i -> Stream Nat i
error during typechecking:
map2
/// clause 1
/// pattern $$i
/// cannot match against deep successor pattern $$i; type: (.[i : Size] -> (Nat -> Nat) -> Stream Nat i -> Stream Nat i)
