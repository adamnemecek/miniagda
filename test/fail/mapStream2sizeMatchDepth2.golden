MiniAgda by Andreas Abel and Karl Mehltretter
--- opening "mapStream2sizeMatchDepth2.ma" ---
--- scope checking ---
--- type checking ---
type  Stream : ++(A : Set) -> - Size -> Set
term  Stream.cons : .[A : Set] -> .[i : Size] -> ^(y1 : A) -> ^(y2 : Stream A i) -> < Stream.cons i y1 y2 : Stream A $i >
type  Nat : Set
term  Nat.zero : < Nat.zero : Nat >
term  Nat.succ : ^(y0 : Nat) -> < Nat.succ y0 : Nat >
term  map2 : .[i : Size] -> (Nat -> Nat) -> Stream Nat i -> Stream Nat i
error during typechecking:
map2
/// clause 1
/// pattern $$i
/// cannot match against deep successor pattern $$i at type .[i : Size] -> (Nat -> Nat) -> Stream Nat i -> Stream Nat i
